### PR TITLE
Generate less initializers in new/upgraded Rails apps (part 2)

### PR DESCRIPTION
**Pull request by ghiculescu**
Follow up to https://github.com/rails/rails/pull/42538 based on some feedback from @dhh:

- Removed `backtrace_silencers.rb`
- Removed `mime_types.rb`

The files just aren't important enough to justify the extra cognitive overload for a new user.

- Removed `wrap_parameters.rb`

I've replaced with this a call in app configuration, and a matching new framework default. Users who were previously using the initializer should keep using it. The default format for new apps is `[:json]`, same as before but without the extra file.

-------------------

You can see all the initializers in a new app here: https://github.com/rails/rails/tree/main/railties/lib/rails/generators/rails/app/templates/config/initializers

### CODE REVIEW

**Review by dhh — APPROVED** [review]

File: railties/lib/rails/application/configuration.rb [hunk 1]
```diff
@@ -237,6 +237,12 @@ def load_defaults(target_version)
 
           if respond_to?(:action_controller)
             action_controller.raise_on_open_redirects = true
+
+            # This can't use the standard configuration pattern because you can call
+            # wrap_parameters at the top level or on any controller.
+            ActiveSupport.on_load(:action_controller) do
```
Comment (rafaelfranca) [inline railties/lib/rails/application/configuration.rb:243]: Just for future references. Things like this should be in Railties. Here we can set a config and use that config in the `ActionController::Railtie` to execute this code. I'm moving in a follow-up commit.

File: railties/lib/rails/application/configuration.rb [hunk 2]
```diff
@@ -237,6 +237,12 @@ def load_defaults(target_version)
...
+              ActionController::Base.wrap_parameters format: [:json]
```
Comment (rafaelfranca) [inline railties/lib/rails/application/configuration.rb:244]: You broke `ActionController::Api` by classing `ActionController::Base` explicitly. Also fixing in a follow-up commit.

File: railties/lib/rails/application/configuration.rb (hunk 1 above)
Comment (ghiculescu) [reply railties/lib/rails/application/configuration.rb:243]: Ahhh, I should have found https://github.com/rails/rails/blob/1b0c914133a48e82832e1017ce9aa51ff021da08/actionpack/lib/action_controller/railtie.rb#L92 as an example.

I can do a followup unless you're already doing it?

Comment (rafaelfranca) [reply railties/lib/rails/application/configuration.rb:243]: I'm already doing.

Comment (rafaelfranca) [reply railties/lib/rails/application/configuration.rb:243]: Fixed in https://github.com/rails/rails/commit/4db42073cdb89157e0494542acb5e3fae6f55d33

**Pull request merged**